content/en/docs/Reference/file-based-catalogs.md

```diff
@@ -0,0 +1,586 @@
+directory.
+
+### Schema
+
+At its core, file-based catalogs use a simple format that can be extended with arbitrary schemas. The format that all
+file-based catalog blobs must adhere to is the `Meta` schema. The below [cue][cuelang-spec] `_Meta` schema defines all
+file-based catalog blobs.
+
+> **NOTE**: No cue schemas listed in this specification should be considered exhaustive. The `opm validate` command has
+> additional validations that are difficult/impossible to express concisely in cue.
+
+```cue
+_Meta: {
+  // schema is required and must be a non-empty string
+  schema: string & !=""
+
+  // package is optional, but if it's defined, it must be a non-empty string
+  package?: string & !=""
+
+  // properties is optional, but if it's defined, it must be a list of 0 or more properties
+  properties?: [... #Property]
+}
+
+#Property: {
+  // type is required
+  type: string & !=""
+
+  // value is required, and it must not be null
+  value: !=null
+}
+```
+
+### OLM-defined schemas
+
+An OLM index currently uses two schemas: `olm.package` and `olm.bundle`, which correspond to OLM's existing package and
+bundle concepts.
+
+Each operator package in an index requires exactly one `olm.package` blob and one or more `olm.bundle` blobs.
+
+> **NOTE**: All `olm.*` schemas are reserved for OLM-defined schemas. Custom schemas must use a unique prefix (e.g. a
+> domain that you own).
+
+#### `olm.package`
+
+An `olm.package` defines package-level metadata for an operator. This includes its name, description, default channel
+and icon.
+
+The `olm.package` [cue][cuelang-spec] schema is:
+```cue
+#Package: {
+  schema: "olm.package"
+  
+  // Package name
+  name: string & !=""
+
+  // A description of the package
+  description?: string
+
+  // The package's default channel
+  defaultChannel: string & !=""
+  
+  // An optional icon
+  icon?: {
+    base64data: string
+    mediatype:  string
+  }
+}
+```
+
+#### `olm.bundle`
+
+<!--
+TODO(joelanford): Add a description of the `olm.bundle` schema here 
+-->
+
+The `olm.bundle` cue schema is:
+```cue
+#Bundle: {
+  schema: "olm.bundle"
+  package: string & !=""
+  name: string & !=""
+  image: string & !=""
+  properties: [...#Property]
+  relatedImages?: [...#RelatedImage]
+}
+
+#Property: {
+  // type is required
+  type: string & !=""
+
+  // value is required, and it must not be null
+  value: !=null
+}
+
+#RelatedImage: {
+  // image is the image reference
+  image: string & !=""
+  
+  // name is an optional descriptive name for an image that
+  // helps identify its purpose in the context of the bundle
+  name?: string & !=""
+}
+```
+
+### Properties
+
+Properties are arbitrary pieces of metadata that can be attached to file-based catalog schemas. The type field is a
+string that effectively specifies the semantic and syntactic meaning of the value field. The value can be any arbitrary
+JSON/YAML.
```

Comment:
I may need to read through the rest of the documentation set, but it may be worth throwing an example here - it wasn't immediately obvious to me when I would need to specify custom properties. It's also entirely possible that we assume some level of familiarity with OLM so this wouldn't be the best place to document usage.

Reply:
This will become _much_ more apparent when we define a new constraint type that allows users to define constraints over arbitrary properties. We don't have that right now though, so I'd hesitate to mention that as an example.

Right now, I don't think there's a specific use case for custom properties.

Not opposed to an example though, if we think it makes sense at this point.